src/model/Stroke.cpp

```diff
@@ -565,35 +565,44 @@ void Stroke::calcSize()
 		Element::height = 0;
 	}
 
-	double minX = points[0].x;
-	double maxX = points[0].x;
-	double minY = points[0].y;
-	double maxY = points[0].y;
+	double minX = DBL_MAX;
+	double maxX = DBL_MIN;
+	double minY = DBL_MAX;
+	double maxY = DBL_MIN;
 
-	for (int i = 1; i < this->pointCount; i++)
+	bool hasPressure = points[0].z != Point::NO_PRESSURE;
+	double halfThick = this->width / 2.0;  //  accommodate for pen width
+
+	for (int i = 0; i < this->pointCount; i++)
 	{
-		if (minX > points[i].x)
+		if (hasPressure) halfThick = points[i].z / 2.0;
+
+		double edge = points[i].x - halfThick;
+		if (minX > edge)
```

Comment:
You can make this cleaner and clearer by using `std::min` and `std::max`:
```suggestion
		minX = std::min(minX, points[i].x - halfThick);
		minY = std::min(minY, points[i].y - halfThick);
		maxX = std::max(maxX, points[i].x + halfThick);
		maxY = std::max(maxY, points[i].y + halfThick);						
```

Reply:
Gee.. why didn't I see that!  Thanks!